keel-sql/src/test/kotlin/com/netflix/spinnaker/keel/sql/SqlDeliveryConfigRepositoryPeriodicallyCheckedTests.kt

```diff
@@ -24,12 +30,63 @@ internal object SqlDeliveryConfigRepositoryPeriodicallyCheckedTests :
   }
 
   override fun flush() {
-    SqlTestUtil.cleanupDb(jooq)
+    cleanupDb(jooq)
   }
 
   @JvmStatic
   @AfterAll
   fun shutdown() {
     testDatabase.dataSource.close()
   }
+
+  fun pausedApplicationTests() = rootContext<Fixture<DeliveryConfig, SqlDeliveryConfigRepository>> {
+    fixture {
+      Fixture(factory, createAndStore, updateOne)
+    }
+
+    after { flush() }
+
+    context("an application is paused") {
+      before {
+        createAndStore(2)
+          .also {
+            SqlPausedRepository(jooq, sqlRetry, clock)
+              .pauseApplication(it.first().application, "fzlem@netflix.com")
+          }
+      }
+
+      test("delivery config is ignored") {
+        expectThat(nextResults()).hasSize(1)
+      }
+    }
+  }
+
+  fun lockExpiryTests() = rootContext<Fixture<DeliveryConfig, SqlDeliveryConfigRepository>> {
+    fixture {
+      Fixture(factory, createAndStore, updateOne)
+    }
+
+    after { flush() }
+
+    context("a delivery config was locked for checking previously") {
+      before {
+        createAndStore(1)
+        nextResults()
+      }
+
+      test("the delivery config is not returned in a check cycle") {
+        expectThat(nextResults()).isEmpty()
+      }
+
+      context("after some time passes") {
+        before {
+          clock.incrementBy(Duration.ofMinutes(35))
+        }
+
+        test("the lock expires") {
+          expectThat(nextResults()).hasSize(1)
+        }
+      }
+    }
+  }
```

Comment:
Test we don't check delivery configs already being checked by another instance, and that we can ignore a stale lock and re-check after a certain amount of time.